source/common/network/connection_impl.cc

```diff
@@ -554,6 +558,7 @@ void ConnectionImpl::onReadReady() {
     return;
   }
 
+  want_read_ = false;
```

Comment:
should we also set this false for the early return onRead above?  If not please comment.

Reply:
Commented.

MockTransportConnectionImplTest.ResumeWhileAndAfterReadDisable covers the early return case which requires us to clear want_read_ just before doRead.

Reply:
Sorry how is want_read_ cleared before onRead above? Maybe update the comment above also as to why we don't need to clear or how it's already cleared?

Reply:
Added comment.